OpenMarket/OpenMarket/Network/NetworkMananger.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  NetworkMananger.swift
+//  OpenMarket
+//
+//  Created by Aaron, Jpush on 2022/11/14.
+//
+
+import Foundation
+
+struct NetworkManager {
+    let successRange = 200..<300
+    
+    enum requestType {
+        case healthChecker
+        case searchProductList(pageNo: Int, itemsPerPage: Int)
+        case searchProductDetail(productNumber: Int)
+    }
+    
+    private func generateURL(type: requestType) -> URL? {
+        let host = "https://openmarket.yagom-academy.kr"
+        
+        switch type {
+        case .healthChecker:
+            let url = URL(string: "healthChecker", relativeTo: URL(string: host))
+            
+            return url?.absoluteURL
+        case .searchProductList(let pageNo, let itemsPerPage):
+            var components = URLComponents(string: "\(host)/api/products")
+            let pageNoParam = URLQueryItem(name: "page_no", value: "\(pageNo)")
+            let itemsPerPageParam = URLQueryItem(name: "items_per_page", value: "\(itemsPerPage)")
+            components?.queryItems = [pageNoParam, itemsPerPageParam]
+            
+            return components?.url
+        case .searchProductDetail(let productNumber):
+            let url = URL(string: "\(productNumber)", relativeTo: URL(string: "\(host)/api/products/"))
+            
+            return url?.absoluteURL
+        }
+    }
+
+    func fetch(type: requestType, completion: @escaping (Completionable) -> Void) {
+        guard let url = generateURL(type: type) else {
+            return
+        }
+        
+        switch type {
+        case .healthChecker:
+            getHealthChecker(url) { statusCode in
+                completion(statusCode)
+            }
+        case .searchProductList(_, _):
+            getProductsList(url) { productsList in
+                completion(productsList)
+            }
+        case .searchProductDetail(_):
+            getProductDetail(url) { product in
+                completion(product)
+            }
+        }
+    }
+    
+    private func getHealthChecker(_ url: URL, completion: @escaping (Int) -> Void) {
```

Comment:
```suggestion
    private func getHealthChecker(_ url: URL, completion: @escaping (StatusCode) -> Void) {
```
`typealias`해둔 StatusCode를  사용하시는건 어떤가요?

Reply:
해당 부분 statusCode로 수정했습니다!